.editorconfig

```diff
@@ -0,0 +1,14 @@
+root = true
+
+[*]
+end_of_line = lf
+charset = utf-8
+trim_trailing_whitespace = true
+insert_final_newline = true
+indent_style = space
+indent_size = 4
+max_line_length = 120
```

Comment:
i think this configs could conflict with rustfmt, so editior might format with something else during autosave and rustfmt might do something else. in general i think rustfmt does a good job of handling this so i don't think this is necessary.

Reply:
This is not for Rust only, but a general way for configuring editors.

We can add an entry to match rust files to match rustfmt but I think it's only one line (max lines?)

Reply:
> This is not for Rust only, but a general way for configuring editors.

yes i know but i don't want multiple ways to configure same thing.

is there any way to apply this settings for anything other than `*.rs` files?

Reply:
Yup, use `unset`. I think the only option to disable is `max_line_length`. The others must be the same

Reply:
problem is not that they are same, its just bad practice to have two conflicting configurations. when we want to update something in rustfmt we would need to update here as well. i would suggest we remove `.editorconfig` completely because code is mostly in rust and only some scripts in js

Reply:
If you want I can unset everything for Rust only. A `.editorconfig` file is rarely (very rarely) touched again in a project so we are pretty safe.
But removing it completely is not a good practice, especially if you don't want any "ide" related files such as .VSCode or .idea

Reply:
> If you want I can unset everything for Rust only.

yes please do

Reply:
Done